Update : adding documentation for Kubeflow Pipelines (#3)
inarix/infra-test-devops-google: docs/notebooks/README.md

# Kubeflow Jupyter notebooks

## What is a Kubeflow Jupyter notebook?

A Kubeflow jupyter notebook is simply a [Jupyter notebook](https://jupyter.org/) handled and created using the kubeflow UI. In this way, we can add some environement variables to it in an automated way, specify some GPU or Memory/CPU limits.

## How to set someone as kubeflow-admin

What is a kubeflow-admin ? This is a Kubeflow Profile that can do/see everything he wants on the kubeflow server. 

> By default this is the one that creates the cluster when deployed with gcp-iap (I nean by that not the way we use to deploy it at Inarix). 

Reviewer: ```suggestion
By default this is the one that creates the cluster when deployed with gcp-iap (I mean: the way we use to deploy it at Inarix). 
```

> In our way, then wee need to add to the admin profile.: 

Reviewer: ```suggestion
Then add to the admin profile: 
```

```yaml
    annotation:
      rbac.authorization.kubeflow.org/aggregate-to-kubeflow-admin: "true"
```

## What is a PodDefault ? 

A PodDefault is the way we have to add some default environment variables to a kubeflow jupyter notebook also addding some predifined volumes like our Inarix Filestore.
A Pod default looks like a simple YAML file like the [one given](notebook_pod_default.yaml) in this repository.

## How to add some labels flags attachment?

> By default, the only way to add some Configuration files for the creation of a jupyter notebook server in the kubeflow ui is to add the ```add-gcp-secret``` label. However in the Kubeflow 1.1 installation files we can give some other labels flags that can be triggered to better fetch some configuration on the kubeflow UI. 

Reviewer: ```suggestion
By default, the only way to add Configuration files for the creation of a jupyter notebook server in the kubeflow ui is to add the ```add-gcp-secret``` label. However in the Kubeflow 1.1 installation files we can give other labels flags that can be triggered to fetch a configuration on the kubeflow UI. 
```

> To do this, open the ```.cache/manifests/manifests-1.1-branch/jupyter/jupyter-web-app/base/configs/spawner_ui_config.yaml``` and in the ```configuration``` section you will find an empty list with some examples in comment.

Reviewer: ```suggestion
To do this, open the ```.cache/manifests/manifests-1.1-branch/jupyter/jupyter-web-app/base/configs/spawner_ui_config.yaml``` and in the ```configuration``` section you will find an empty list with examples in comment.
```

```yaml
  configurations:
    # List of labels to be selected, these are the labels from PodDefaults
    # value:
    #   - add-gcp-secret
    #   - default-editor
    value: []
    readOnly: false
```

> Simply add the wanted selector labels as you want before deploying kubeflow or simply modify it and re-apply the all Kfctl configuration file.

Reviewer: ```suggestion
Add the wanted selector labels as you want before deploying kubeflow or modify it and re-apply the all kfctl configuration files.
```
